src/doc/DocHighlightThread.js

```diff
@@ -555,47 +555,25 @@ class DocHighlightThread extends AnnotationThread {
         // Delete matching comment from annotation
         this.comments = this.comments.filter(({ id }) => id !== annotationIDToRemove);
 
-        if (this.type === TYPES.highlight_comment && this.comments.length < 0) {
+        if (this.type === TYPES.highlight_comment && this.comments.length <= 0) {
             this.type = TYPES.highlight;
 
-            // If the user doesn't have permission to delete the entire highlight
-            // annotation, display the annotation as a plain highlight
-            if (!this.canDelete) {
+            if (this.canDelete) {
+                this.delete({ id: this.id });
+                this.destroy();
+            } else {
+                // If the user doesn't have permission to delete the entire highlight
```

Comment:
Question about this logic: Why don't we just ALWAYS turn them into plain highlights? It seems quite inconsistent if we have some cases that do, and others that just completely delete it

Reply:
This was a decision made by product a while back. 

I could also see that causing problems if a developer has only highlight comments disabled, then we're creating plain highlights on their file when that's been disabled for them.